### ISSUE TITLE
Allow multi-site operators in `OpSum`

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#741]
Allow multi-site operators in `OpSum` as well operator-dependent keyword arguments. For example:
```julia
os = OpSum()
os += 0.5, "CX", 1, 3
```
This would make it a bit easier to define Hamiltonians for forming MPOs as well as a Trotter-Suzuki decomposition interface that @GTorlai is working on for PastaQ.

For the sake of making MPOs, it could be implemented by first forming the operator and then doing an expansion in a local operator basis in order to transform it into a sum of products of local operators, but we could still add support for this in the `OpSum` data structure before the MPO constructor support is implemented so it can be used for other applications.

In addition, it would be helpful to allow adding keyword arguments for operators with parameters, for example:
```julia
os += 1.2, "CRz", 1, 4, (ϕ = π/3,)
```

**Comment by emstoudenmire** [comment]
This feature could potentially take advantage of issue #722 when that is implemented. Then one could, say, preprocess "CX" into a sum of products of operators A1*A3 + B1*B3 +... but where now the A, B, etc. operators can be any matrix. So then this feature would not have to rely on a complete basis of operators being defined for the site type and/or could use an SVD to split the operator.

**Comment by mtfishman** [comment]
Closed by #749.
